### ISSUE TITLE
Bug: only the last connected flow node is executed instead of all connected flow nodes

### ISSUE THREAD

**Issue by hybridherbst** [KhronosGroup/glTF-InteractivityGraph-AuthoringTool#17]
In this simple graph, only "world" is logged – my expectation is that "hello" and "world" are logged instead:

I could not find details in the spec on what happens when the same output flow socket is connected to multiple input flow sockets. I did find that there's a `flow/sequence` node but would not expect that to be necessary. My understanding is that the one value output can also be connected to multiple value inputs. cc @lexaknyazev, I might have missed a place where that is defined?

<img width="1181" alt="image" src="https://github.com/user-attachments/assets/d0256af6-43c1-41ad-81c4-58e9ce7a1def">

```json
{
	"nodes": [
		{
			"type": "event/onStart",
			"values": [],
			"configuration": [],
			"flows": [
				{
					"id": "out",
					"node": 2,
					"socket": "in"
				},
				{
					"id": "out",
					"node": 1,
					"socket": "in"
				}
			],
			"metadata": {
				"positionX": "271.7791644049969",
				"positionY": "274.00562447450295"
			}
		},
		{
			"type": "ADBE/output_console_node",
			"values": [
				{
					"id": "message",
					"value": "world",
					"type": 7
				}
			],
			"configuration": [],
			"flows": [],
			"metadata": {
				"positionX": "806.8545094544273",
				"positionY": "474.4824028307539"
			}
		},
		{
			"type": "ADBE/output_console_node",
			"values": [
				{
					"id": "message",
					"value": "hello",
					"type": 7
				}
			],
			"configuration": [],
			"flows": [],
			"metadata": {
				"positionX": "808.3572132317787",
				"positionY": "185.60156250000006"
			}
		}
	],
	"variables": [],
	"customEvents": [],
	"types": [
		{
			"signature": "bool"
		},
		{
			"signature": "int"
		},
		{
			"signature": "float"
		},
		{
			"signature": "float2"
		},
		{
			"signature": "float3"
		},
		{
			"signature": "float4"
		},
		{
			"signature": "float4x4"
		},
		{
			"signature": "custom",
			"extensions": {
				"AMZN_interactivity_string": {}
			}
		}
	]
}
```

**Comment by hybridherbst** [comment]
This works:
<img width="601" alt="image" src="https://github.com/user-attachments/assets/f0c65c97-a3b3-4fb8-9404-0e292261adde">

So either the bug here is that all connected flows should be run, or the bug is that I can connect the outputs to more than one thing at all, and/or that there is no log telling me that I messed up.

**Comment by lexaknyazev** [comment]
An output flow cannot be connected to more than one input flow.

To have multiple output ops from the same node, use `flow/sequence` or chain the `out` output flows.

**Comment by hybridherbst** [comment]
Thank you for the clarification! 
Could you point me at the part of the spec where this is stated? I wasn't able to find it.

**Comment by lexaknyazev** [comment]
It's impossible by design given the JSON structure like
```json
  {
    "type": "event/onStart",
    "flows": [
      {
        "id": "out",
        "node": 0,
        "socket": "in"
      }
    ]
  }
```
and this spec language:
> The same `id` value **MUST NOT** be used more than once within each of the `configuration`, `values`, and `flows` arrays.

An explicit note won't hurt, of course.

**Comment by boguscoder** [comment]
@lexaknyazev  / @hybridherbst  / @mattmacf98  I cant find wording in new draft that precludes 1:N or N:1 connections for flows, could you please clarify where its not allowed

**Comment by mattmacf98** [comment]
The prohibition of 1:N connections is implicit in the structure of the nodes. Since the node JSON defines the out flow sockets as key value pairs, It is implicitly enforced via JSON semantics that an out flow can only point to one node but an in flow may be pointed to by many nodes

e.x. we cannot have a single out flow point to N in flows (1:N)

```
{
   flows: {
     out: {node: 0, socket: "in"},
     out: {node: 1, socket: "in}
   }
}
``` 

A JSON parser would toss out the first flow.out in favor of the second.

on the other hand, a single input flow can be pointed to by many output flows (N:1)

It may be worth explicitly calling this out if it is not clear what a "normative" JSON parser would do in the above scenario

**Comment by boguscoder** [comment]
right, thanks, I was still living mentally in previous iteration where flows were arrayed
